### DISCUSSION TITLE
Hydrate Property is called before Booted

### DISCUSSION THREAD

**Discussion by xxdalexx** [livewire/livewire#5290] [1. Help]
[Documentation States:](https://laravel-livewire.com/docs/2.x/lifecycle-hooks) that booted
> Runs on every request, after the component is mounted or hydrated, but before any update methods are called

However, (running on 2.10.7) doing a simple test of placing a bunch of dd()'s and removing them, the lifecycle seems to run in hydrateProperty, hydrate, then booted.

**Comment by joshhanley** [comment]
@xxdalexx that is correct, as specific it runs "after the component is mounted or hydrated".

The `hydrate` and `hydrateFoo` property methods are part of the hydration process, hence why `booted` runs after them.

Hope this helps!

**Reply by xxdalexx** [reply]
Yeah, I realized I had a lapse in my reading ability combined with thinking the list order in the docs was the lifecycle order, but wasn't able to come back and delete this. I had figured it was buried and forgotten by now :).

Sorry for the waste of time.

**Reply by joshhanley** [reply]
@xxdalexx all good! Yeah I've just been going back through my notifications 😬 Good to leave it here any case anyone else wonders the same thing 🙂
